### DISCUSSION TITLE
HTTP Response Status Line

### DISCUSSION THREAD

**Discussion by scrumgelist** [nextgenhealthcare/connect#5589] [Feature/Enhancement Ideas]
Within the **HTTP Listener** setup it is possible to set the response status code.

It is not possible to set the accompanying status text. (I'd love to wrong)

![image](https://user-images.githubusercontent.com/99478435/207962065-a69937a7-5473-43d2-8c61-8b0fb5a9492b.png)

Please either allow the text to be set along with the code or provide an adjacent box for the text.

If done ${customResponseCode} could contain "540 Patient not found" instead of "540" or the adjecent box the text element "Patient not found".  This enables the simple sending of bilaterally agreed status messages.

Out of respect for 418 and 451 it may be desirable to prevent the changing of 'official code texts' if the selected status code has an official meaning, so long as the offical text is attached to the status code if specified.

See discussion re need for request here: https://github.com/nextgenhealthcare/connect/discussions/5587

**Comment by tom08zehn** [comment]
I'd like to share my idea in order to minimize GUI and behavior changes, the modified code could:

* First, split the string in "Response Status Code" field (actually `connectorProperties.getResponseStatusCode()`) by space char ([here](https://github.com/nextgenhealthcare/connect/blob/master/server/src/com/mirth/connect/connectors/http/HttpReceiver.java#L439))
* Then do the [same as today](https://github.com/nextgenhealthcare/connect/blob/master/server/src/com/mirth/connect/connectors/http/HttpReceiver.java#L440) _with the first item_ of the resulting array (try to convert to int etc.)
* Finally, if array length > 1 then use the rest as response status message ([here](https://github.com/nextgenhealthcare/connect/blob/master/server/src/com/mirth/connect/connectors/http/HttpReceiver.java#L457) and [here](https://github.com/nextgenhealthcare/connect/blob/master/server/src/com/mirth/connect/connectors/http/HttpReceiver.java#L503)), joined again with space char; if not then do the same as today (simply use status code as status message)

Example: string "540 Patient not found" results in status code 540 and status message "Patient not found" and both then is returned in the HTTP servlet response

---

Optionally, a response transformer could get 2 new reserved variables, similar to `responseStatusMessage`, to be able to do the following in a HTTP Sender connector's response transformer:

```js
httpResponseStatusCode = 540;
httpResponseStatusMessage = "Patient not found";
```

Means:

* If only `httpResponseStatusCode` is set then set then use `httpResponseStatusCode.toString()` as response status code _and_ message
* If both values are set for `httpResponseStatusCode` _and_ `httpResponseStatusMessage` then use httpResponseStatusCode.toString() as response status code and `httpResponseStatusMessage` as status message

A tiny documentation change would be needed (see [here](https://docs.nextgen.com/bundle/Mirth_User_Guide_40/page/connect/connect/topics/c_HTTP_Listener_connect_ug.html), item "K").

**Comment by wi4seo** [comment]
Hi Guys,

FREE Registration for Mirth Unconference 2023 is Now Opened!

To know more, please visit the event page and register yourself: https://wi4.org/mirth-connect-unconference-feb-2023/

Stay Tuned for More Updates!​

​Thanks & Regards
